py/desispec/scripts/tsnr_afterburner.py

```diff
@@ -0,0 +1,197 @@
+                        help = 'Dir. to write to.')
+    parser.add_argument('--prod', type = str, default = None, required=True,
+                        help = 'Path to reduction, e.g. /global/cfs/cdirs/desi/spectro/redux/blanc/')
+    parser.add_argument('--camera', type = str, default = None, required=False,
+                        help = 'Camera to reduce.')
+    parser.add_argument('--summary_only', type = int, default = 0, required=False,
+                        help = 'Write only summary file.')
+    parser.add_argument('--expids', type = str, default = None, required=False,
+                        help = 'Comma separated list of exp ids to be reduced.')
+    args = None
+    if options is None:
+        args = parser.parse_args()
+    else:
+        args = parser.parse_args(options)
+    return args
+
+args=parse()
+
+#
+log = get_logger()
+
+# Set SPEC_PROD also.
+calibdir = os.environ['DESI_SPECTRO_CALIB']
+
+log.info('outdir = {}'.format(args.outdir))
+log.info('prod = {}'.format(args.prod))
+log.info('camera = {}'.format(args.camera))
+log.info('summary_only = {}'.format(args.summary_only))
+log.info('expids = {}'.format(args.expids))
+
+log.info('$DESI_SPECTRO_CALIB = {}'.format(calibdir))
+
+petals = np.arange(10).astype(str)
+
+if args.camera is None:
+    cameras = [x[0] + x[1] for x in itertools.product(['b', 'r', 'z'], petals.astype(np.str))]
+else:
+    cameras = [args.camera]
+
+if args.expids is not None:
+    expids = [np.int(x) for x in args.expids.split(',')]
+else:
+    expids = None
+    
+# 
+cframes = {}
+
+for cam in cameras:
+    cframes[cam] = list(glob.glob('{}/exposures/*/*/cframe-{}-*.fits'.format(args.prod, cam)))
+
+sci_frames = {}
+
+for cam in cameras:
+    sci_frames[cam] = []
+    
+    for cframe in cframes[cam]:
+        hdul = fits.open(cframe)
+        hdr  = hdul[0].header 
+        
+        flavor = hdr['FLAVOR']
+        prog = hdr['PROGRAM']
+        expid = hdr['EXPID']
+        
+        if expids is not None:
+            if expid not in expids:
+                continue
+            
+        if flavor == 'science':
+            sci_frames[cam].append(cframe)
+
+        hdul.close()
+        
+    log.info('{} science frames to reduce for {}.'.format(len(sci_frames[cam]), cam))
+    
+for cam in cameras:
+    summary  = None
+    
+    for kk, x in enumerate(sci_frames[cam]):
+        hdul = fits.open(x)
+        hdr  = hdul[0].header
+
+        flavor = hdr['FLAVOR']
+        prog = hdr['PROGRAM']
+        
+        parts = prog.split(' ')
+
+        if parts[0] == 'SV1':
+            parts  = x.split('/')
+                
+            night  = parts[9]
+            expid  = np.int(parts[10])
+
+            name   = parts[-1]
+
+            parts  = name.split('-')
+            camera = parts[1]
+            
+            calib  = findfile('fluxcalib', night=night, expid=expid, camera=camera, specprod_dir=None)
+            calib  = calib.replace('SPCALIB', calibdir)
```

Comment:
What is the case where this SPCALIB replacement is needed?  I recognize that from header value shortening, but findfile should find the full correct path to the flux calibration file without a 'SPCALIB' in it, and `calibdir = os.environ['DESI_SPECTRO_CALIB']` only has a default fluxcalib used by nightwatch, not the nightly fluxcalib files for this night, expid.  Since `calib` shouldn't have 'SPCALIB` in it, this is a no-op, but could lead to confusion about what the code is trying to do.